src/pam/rpassword.rs

```
@@ -102,13 +102,10 @@ fn read_unbuffered(source: &mut dyn io::Read) -> io::Result<PamBuffer> {
     Ok(password)
 }
 
-const BACKSPACE: u8 = 8;
-
 fn erase_feedback(sink: &mut dyn io::Write, i: usize) {
+    const BACKSPACE: u8 = 0x08;
     for _ in 0..i {
-        if sink.write(&[BACKSPACE, b' ', BACKSPACE]).is_err() {
-            return;
-        }
+        let _ = sink.write(&[BACKSPACE, b' ', BACKSPACE]);
```

The early return was intentional. It would likely error on the next write too and the original sudo code also does an early return.

---

I have no objections to that, but that does violate the stated invariant, so I'd prefer the "ignore all errors" version.

---

(And it makes little sense to me to only do the early return when we're doing typing a password, but *not* do an early return when we're sending a line kill--as the original code appears to do--that feels a bit confused)